# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1356307346 18000
#      Sun Dec 23 19:02:26 2012 -0500
# Branch classdef
# Node ID 896cf57695379ac4bf959ec93277da938b6a5a2d
# Parent  5861c4bde387f340c30b3db9a01019de13d6cc7a
Add visit_funcall method to JIT tree walker.

* libinterp/interp-core/pt-jit.h (jit_convert::visit_funcall): New method.
* libinterp/interp-core/pt-jit.cc (jit_convert::visit_funcall): Likewise.

diff --git a/libinterp/interp-core/pt-jit.cc b/libinterp/interp-core/pt-jit.cc
--- a/libinterp/interp-core/pt-jit.cc
+++ b/libinterp/interp-core/pt-jit.cc
@@ -650,16 +650,22 @@ jit_convert::visit_constant (tree_consta
 
 void
 jit_convert::visit_fcn_handle (tree_fcn_handle&)
 {
   throw jit_fail_exception ();
 }
 
 void
+jit_convert::visit_funcall (tree_funcall&)
+{
+  throw jit_fail_exception ();
+}
+
+void
 jit_convert::visit_parameter_list (tree_parameter_list&)
 {
   throw jit_fail_exception ();
 }
 
 void
 jit_convert::visit_postfix_expression (tree_postfix_expression& tpe)
 {
diff --git a/libinterp/interp-core/pt-jit.h b/libinterp/interp-core/pt-jit.h
--- a/libinterp/interp-core/pt-jit.h
+++ b/libinterp/interp-core/pt-jit.h
@@ -122,16 +122,18 @@ public:
   void visit_multi_assignment (tree_multi_assignment&);
 
   void visit_no_op_command (tree_no_op_command&);
 
   void visit_constant (tree_constant&);
 
   void visit_fcn_handle (tree_fcn_handle&);
 
+  void visit_funcall (tree_funcall&);
+
   void visit_parameter_list (tree_parameter_list&);
 
   void visit_postfix_expression (tree_postfix_expression&);
 
   void visit_prefix_expression (tree_prefix_expression&);
 
   void visit_return_command (tree_return_command&);
 
